### PR TITLE
Fix Poco::Process::launch (UNIX) - memory leak when launching invalid command 

### PR DESCRIPTION
**Pull request by pavledragisic**

### CODE REVIEW

File: Foundation/src/Process_UNIX.cpp [hunk 1]
```diff
@@ -182,76 +182,83 @@ ProcessHandleImpl* ProcessImpl::launchByForkExecImpl(const std::string& command,
...
+			if (errPipe) dup2(errPipe->writeHandle(), STDERR_FILENO);
+			if (outPipe) outPipe->close(Pipe::CLOSE_BOTH);
+			if (options & PROCESS_CLOSE_STDOUT) close(STDOUT_FILENO);
+			if (errPipe) errPipe->close(Pipe::CLOSE_BOTH);
+			if (options & PROCESS_CLOSE_STDERR) close(STDERR_FILENO);
+			// close all open file descriptors other than stdin, stdout, stderr
+			long fdMax = sysconf(_SC_OPEN_MAX);
+			// on some systems, sysconf(_SC_OPEN_MAX) returns a ridiculously high number
+			if (fdMax > CLOSE_FD_MAX) fdMax = CLOSE_FD_MAX;
+			for (long i = 3; i < fdMax; ++i)
```
Comment (aleks-f) [inline Foundation/src/Process_UNIX.cpp:245]: @pavledragisic let's also fix this CodeQL alert

File: Foundation/src/Process_UNIX.cpp [hunk 2]
```diff
@@ -182,76 +182,83 @@ ProcessHandleImpl* ProcessImpl::launchByForkExecImpl(const std::string& command,
...
-	argv[i++] = const_cast<char*>(command.c_str());
-	for (const auto& a: args)
+	do
 	{
-		argv[i++] = const_cast<char*>(a.c_str());
-	}
-	argv[i] = NULL;
-
-	const char* pInitialDirectory = initialDirectory.empty() ? 0 : initialDirectory.c_str();
+		// We must not allocated memory after fork(),
```
Comment (aleks-f) [inline Foundation/src/Process_UNIX.cpp:187]: spelling: alllocate~~d~~

**Review by aleks-f — CHANGES_REQUESTED** [review]
LGTM

File: Foundation/src/Process_UNIX.cpp (hunk 2 above)
Comment (pavledragisic) [reply Foundation/src/Process_UNIX.cpp]: Fixed in [8c4d83c](https://github.com/pocoproject/poco/pull/4267/commits/8c4d83c86860d1730a71d29b58aae1e9d8080db0)

File: Foundation/src/Process_UNIX.cpp (hunk 1 above)
Comment (pavledragisic) [reply Foundation/src/Process_UNIX.cpp]: Fixed in [8c4d83c](https://github.com/pocoproject/poco/pull/4267/commits/8c4d83c86860d1730a71d29b58aae1e9d8080db0)

**Pull request merged**